### ISSUE TITLE
[Bug]  Discord cookies are not deleted after enabling local storage cleanup option. 

### ISSUE THREAD

**Issue by ghost** [mrdokenny/Cookie-AutoDelete#1263]
### Acknowledgements

- [X] I acknowledge that I have read the above items

### Describe the bug

1. Go to discord.com and make a account and login 
2. Close the tab and again go to discord.com 
3. Click on the login button and it will automatically detect your account. 

### To Reproduce

Same as Describe the bug 

### Expected Behavior

I expected that discord wouldn't recognize my old account. 

### System Info - Operating System (OS)

Arch Linux linux 5.14.16.arch1-1

### System Info - Browser Info

Firefox 94.0.1 (x64) latest 

### System Info - CookieAutoDelete Version

3.6.0

**Comment by kennethtran93** [comment]
What is your Firefox privacy setting at?

What browsing data types have you enabled for cleanup?

**Comment by ghost** [comment]
> What is your Firefox privacy setting at?
> 
> What browsing data types have you enabled for cleanup?

I have hardened my firefox for privacy. I use arkenfox user.js - https://github.com/arkenfox/user.js/ 
and I have enabled all the options for cleaup in cookie auto delete extension for maximum privacy. 
![image](https://user-images.githubusercontent.com/69617708/141257647-5e8c3307-dcf3-4e2d-a3ae-ec34f95f786c.png)

**Comment by ghost** [comment]
This is my CAD settings .json file - 
{
  "settings": [
    {
      "name": "activeMode",
      "value": true
    },
    {
      "name": "cacheCleanup",
      "value": true
    },
    {
      "name": "cleanCookiesFromOpenTabsOnStartup",
      "value": true
    },
    {
      "name": "cleanExpiredCookies",
      "value": true
    },
    {
      "name": "contextMenus",
      "value": true
    },
    {
      "name": "contextualIdentities",
      "value": false
    },
    {
      "name": "contextualIdentitiesAutoRemove",
      "value": true
    },
    {
      "name": "debugMode",
      "value": true
    },
    {
      "name": "delayBeforeClean",
      "value": 5
    },
    {
      "name": "discardedCleanup",
      "value": true
    },
    {
      "name": "domainChangeCleanup",
      "value": true
    },
    {
      "name": "enableGreyListCleanup",
      "value": true
    },
    {
      "name": "enableNewVersionPopup",
      "value": false
    },
    {
      "id": "DEPRECATED - use default expressions",
      "name": "greyCleanLocalstorage",
      "value": false
    },
    {
      "name": "indexedDBCleanup",
      "value": true
    },
    {
      "name": "keepDefaultIcon",
      "value": false
    },
    {
      "id": "DEPRECATED - use localStorageCleanup",
      "name": "localstorageCleanup",
      "value": true
    },
    {
      "name": "localStorageCleanup",
      "value": true
    },
    {
      "name": "manualNotifications",
      "value": false
    },
    {
      "name": "notificationOnScreen",
      "value": 3
    },
    {
      "name": "pluginDataCleanup",
      "value": true
    },
    {
      "name": "serviceWorkersCleanup",
      "value": true
    },
    {
      "name": "showNotificationAfterCleanup",
      "value": false
    },
    {
      "name": "showNumOfCookiesInIcon",
      "value": true
    },
    {
      "name": "sizePopup",
      "value": 16
    },
    {
      "name": "sizeSetting",
      "value": 16
    },
    {
      "name": "statLogging",
      "value": false
    },
    {
      "id": "DEPRECATED - use default expressions",
      "name": "whiteCleanLocalstorage",
      "value": false
    }
  ]
}

**Comment by kennethtran93** [comment]
Are you using containers / contextual identities (No Container, Personal, etc)?  If I read the privacy configurations properly, it enables containers.  If you are using containers, you'll need to enable the containers setting in CAD, otherwise we only clean up items in the no containers group.

Also from what you linked, it appears to block all third-party cookies, which _should_ be fine.  As long as it is not the dynamic state partitioning (cookieBehaviour = 5) or cross-site cookies.

I'm presuming that you waited for at least 5 seconds / CAD notification before opening it up again, per your configuration.

**Comment by ghost** [comment]
> Are you using containers / contextual identities (No Container, Personal, etc)? If I read the privacy configurations properly, it enables containers. If you are using containers, you'll need to enable the containers setting in CAD, otherwise we only clean up items in the no containers group.
> 
> Also from what you linked, it appears to block all third-party cookies, which _should_ be fine. As long as it is not the dynamic state partitioning (cookieBehaviour = 5) or cross-site cookies.
> 
> I'm presuming that you waited for at least 5 seconds / CAD notification before opening it up again, per your configuration.

I enabled container setting in CAD but still it is not working. Discord cookies are not getting deleted. It is working fine with other sites. Disabling the container still doesn't work. 
![image](https://user-images.githubusercontent.com/69617708/141409113-d2c93248-a434-470b-ad6d-695ac84a4ab2.png)

**Comment by kennethtran93** [comment]
Apologies for the delay in responding.  As I don't use the arkenfox user.js configuration, it will take me some time to look into this, and will continue to do so after getting some other bugfixes resolved first.

**Comment by ghost** [comment]
No problem. Take ur time :)

**Comment by ItsOnlyBinary** [comment]
I think im having a similar issue to this with ebay.co.uk, after closing ebay.co.uk i get notifications for 3 domains i have whitelisted to allow 1/2 cookies but not all (just auth cookies) but no data is actually cleared for ebay.co.uk

ebay.co.uk has no whitelist or greylist item and is in a normal tab, just restarted my browser and tested with only ebay.co.uk tab and never get the notification or log entry of it closing, appears the issue maybe somewhere in tab closing detection

**Comment by ghost** [comment]
> I think im having a similar issue to this with ebay.co.uk, after closing ebay.co.uk i get notifications for 3 domains i have whitelisted to allow 1/2 cookies but not all (just auth cookies) but no data is actually cleared for ebay.co.uk
> 
> ebay.co.uk has no whitelist or greylist item and is in a normal tab, just restarted my browser and tested with only ebay.co.uk tab and never get the notification or log entry of it closing, appears the issue maybe somewhere in tab closing detection

Yes exactly I am getting a lot of websites where auto cookie delete is not able to delete the cookies properly. So basically u can sign in to that website easily without logging in. :(

**Comment by kennethtran93** [comment]
@ItsOnlyBinary @prorounak2005 what browser are you using?

**Comment by ghost** [comment]
> @ItsOnlyBinary @prorounak2005 what browser are you using?

I am using firefox. I am using arch linux so I have the bleeding edge firefox installed on arch.

**Comment by ItsOnlyBinary** [comment]
Firefox Win10

It does seem some what intermittent as i have two computers with a copy of the same firefox profile, and one was cleaning ebay.co.uk ok and the other was not

**Comment by kennethtran93** [comment]
Ah if you also have the cross-site cookie option/strict enabled, I am still working on updating the webextension to clean those as well.

**Comment by ghost** [comment]
> Ah if you also have the cross-site cookie option/strict enabled, I am still working on updating the webextension to clean those as well.


Actually I couldn't find the cross site cookie option in the webextension but In firefox it is set to strict.

![image](https://user-images.githubusercontent.com/69617708/152633701-0c2f87af-79e7-4bb1-aa1b-21b3231b4b06.png)

**Comment by kennethtran93** [comment]
> > Ah if you also have the cross-site cookie option/strict enabled, I am still working on updating the webextension to clean those as well.
> 
> Actually I couldn't find the cross site cookie option in the webextension but In firefox it is set to strict.
> 
> ![image](https://user-images.githubusercontent.com/69617708/152633701-0c2f87af-79e7-4bb1-aa1b-21b3231b4b06.png)

Cross-site cookie is the all third-party cookies option as well (same as strict mode).

What you may be experiencing is already noted - this is a work in progress.  See #807.
